numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2383,13 +2539,12 @@ NPY_NO_EXPORT PyTypeObject PyGenericArrType_Type = {
 static void
 void_dealloc(PyVoidScalarObject *v)
 {
-    _dealloc_cached_buffer_info((PyObject *)v);
-
     if (v->flags & NPY_ARRAY_OWNDATA) {
         npy_free_cache(v->obval, Py_SIZE(v));
     }
     Py_XDECREF(v->descr);
     Py_XDECREF(v->base);
+    _dealloc_cached_buffer_info((PyObject *)v);
```

Comment:
Any reason for the order change here?

Reply:
No, an oversight because I split this out from a larger change.

Reply:
Fixed those (and squashed with the previous commit).